torch/nn/quantized/modules/linear.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+from ...modules.module import Module
+from ...._jit_internal import weak_module
+
+
+
+@weak_module
+class Linear(Module):
+    r"""
+    A module that wraps the quantized fbgemm linear operator function
+    We adopt the same interface as `torch.nn.Linear`, please see https://pytorch.org/docs/stable/nn.html#torch.nn.Linear
+    for documentation.
+
+    Similar to `torch.nn.Linear`, attributes will be randomly initialized at
+        module creation time and will be overwritten later
+
+    Attributes:
+        _packed_weight: the weight that is first quantized and then packed using
+            fbgemm_linear_pack function
+        bias:   the quantized bias
+        output_scale: `scale` parameter of output Quantized Tensor
+        output_zero_point: `zero_point` parameter for output Quantized Tensor
+    """
+
+    def __init__(self, in_features, out_features, bias=True):
+        super(Linear, self).__init__()
+        weight = torch.randn(out_features, in_features, dtype=torch.float32)
+        weight = torch.quantize_linear(weight, 1.0, 0, torch.qint8)
+        _packed_weight = torch.ops.quantized.fbgemm_linear_prepack(weight)
+
+        output_scale = 1.0
```

Comment:
hm, do we have a valid case where we'd want to initialize Linear from scratch like this? why not take existing Linear module in constructor together with qparams?

Reply:
reason is we want consistency between APIs and workflows in `torch.nn.Module`, `torch.nn.quantized` and modules used for qat.

Because in `torch.nn.Module` we initialize from scratch, and either load the weights or train, we are going to do the same for quantization flow, after we initialize, we will either load from saved quantized weights, or in the case of training we probably load from a checkpoint, and then do qat from there.